cmd/inventory/utils_openstack.go

```diff
@@ -201,232 +197,94 @@ func newOpenStackProviderClient(
 	return gophercloudconfig.NewProviderClient(ctx, authOpts)
 }
 
-// configureOpenStackComputeClientsets configures the OpenStack Compute API clientsets.
-func configureOpenStackComputeClientsets(ctx context.Context, conf *config.Config) error {
-	for _, clientConfig := range conf.OpenStack.Services.Compute {
-		creds := clientConfig.UseCredentials
-		namedCreds := conf.OpenStack.Credentials[creds]
+func configureOpenStackServiceClientset(
+	ctx context.Context,
+	serviceName string,
+	clientset *registry.Registry[openstackclients.ClientScope, openstackclients.Client[*gophercloud.ServiceClient]],
+	serviceConfig config.OpenStackServiceCredentials,
+	conf *config.Config,
+	serviceFunc func(
+		providerClient *gophercloud.ProviderClient,
+		eo gophercloud.EndpointOpts) (*gophercloud.ServiceClient, error)) error {
+
+	// set of credentialNames to create clients from
```

Comment:
Why do we need this one? We should iterate over the list of credentials directly, IMO. Duplicate credentials should probably be considered a configuration issue.